File: app/src/main/java/com/attendance/data/source/local/DBHelper.java
package com.attendance.data.source.local;

import android.content.Context;
import android.database.sqlite.SQLiteDatabase;

import com.attendance.data.DaoMaster;

/**
 * Created by peiqin on 5/12/2017.
 */

public class DBHelper {

    private final static String dbName = "attendance";
    private static DBHelper mInstance;
    private DaoMaster.DevOpenHelper openHelper;
    private Context context;

    public DBHelper(Context context) {
        this.context = context;
        openHelper = new DaoMaster.DevOpenHelper(context, dbName, null);
    }

    public static DBHelper getInstance(Context context) {
        if (mInstance == null) {
            synchronized (DBHelper.class) {
                if (mInstance == null) {
                    mInstance = new DBHelper(context);
                }
            }
        }
        return mInstance;
    }

    public SQLiteDatabase getReadableDatabase() {
        if (openHelper == null) {
            openHelper = new DaoMaster.DevOpenHelper(context, dbName, null);
        }
        SQLiteDatabase db = openHelper.getReadableDatabase();
        return db;
    }

    public SQLiteDatabase getWritableDatabase() {
        if (openHelper == null) {
            openHelper = new DaoMaster.DevOpenHelper(context, dbName, null);
        }
        SQLiteDatabase db = openHelper.getWritableDatabase();
        return db;
    }

}

File: app/src/main/kotlin/com.attendance/data/source/local/DBHelper.kt
package com.attendance.utils

import android.content.Context
import android.database.sqlite.SQLiteDatabase
import android.database.sqlite.SQLiteOpenHelper

class DBHelper private constructor(context: Context, name: String, version: Int) : SQLiteOpenHelper(context, name, null, version) {

    override fun onCreate(db: SQLiteDatabase) {
        try {
            val create_course_table_sql = "create table course(id integer primary key, name varchar(60) not null);"
            db.execSQL(create_course_table_sql)
        } catch (e: Exception) {
            e.printStackTrace()
        }

    }

    override fun onUpgrade(db: SQLiteDatabase, oldVersion: Int, newVersion: Int) {
        when (newVersion) {
            2 -> {
            }
            else -> {
            }
        }
        val drop_course_table_sql = "drop table if exists course;"
        db.execSQL(drop_course_table_sql)
        onCreate(db)
    }

    companion object {
        private var dbHelper: DBHelper? = null

        fun getInstance(context: Context): SQLiteDatabase {
            if (dbHelper == null) {
                dbHelper = DBHelper(context, "attendance", 1)
            }
            return dbHelper!!.readableDatabase
        }
    }

}
